http/org_service.go

```diff
@@ -426,6 +430,85 @@ func newOrganizationLogResponse(id influxdb.ID, es []*influxdb.OperationLogEntry
 	}
 }
 
+// SecretService connects to Influx via HTTP using tokens to manage secrets.
+type SecretService struct {
+	Client *httpc.Client
+}
+
+// LoadSecret is not implemented for http
+func (s *SecretService) LoadSecret(ctx context.Context, orgID influxdb.ID, k string) (string, error) {
+	return "", &influxdb.Error{
```

Comment:
feels like we should  panic here? not being implemented then it is called that would qualify (in my head) as an exception (panic). Does that make sense?

Reply:
Not a bad suggestion. But I think panic is actually a handling.

In go, I usually prefer handle error, including panics, in one last step. 
It's easier for testing, also easier to track all the panics in our system.

Reply:
The point I was driving at, is regardless the language really, an exception is outside the domain the code was implemented for. In those situations, typically we would throw an exception(a panic in go). 

My other reason for a `panic` is that testing code that does is not implemented doesn't make a lot of sense. It should not be used, and if it is then it should panic to indicate to the consumer this is not valid. An error being returned can send the next person who might incorporate the secret service down a rabbit hole. We can avoid that wasted effort with a panic to indicate clearly that the behavior is not implemented.  

Probably shoudl have just wrote all that the first time lol

Reply:
I kinda agree your point actually. However we have similar pattern in somewhere else, like 
https://github.com/influxdata/influxdb/blob/1f1b1c492e502ca834cae6305e55e523edb81253/task/options/flux_deps.go#L35 The benefit is giving the consumer more flexibility to override and modify.  

I think for a long term wise, we could revisit our dependency strategy, like split the service, so everything make senses. Not just for secret, but for other crud services as well. For now, I think both works, I personally prefer to keep it this way.

Reply:
the points you're making don't really fall in line with what I'm getting at. If you want to leave it as is, go for it. We need to have a definition for these situations as a company, sooner than later. The more  unique each place is, the worse off the code base is generally.